### DISCUSSION TITLE
Launch Airflow Webserver via multiprocessing

### DISCUSSION THREAD

**Discussion by da-tubi** [apache/airflow#30724] [General]
### Description

https://github.com/apache/airflow/blob/main/airflow/cli/commands/webserver_command.py

Currently, Airflow webserver is launched via subprocess. It requires a Python virtual environment (with gunicorn installed) to launch the Airflow webserver. For people who want to use pex to deploy Airflow, it is a blocking issue.

### Use case/motivation

Here is the use cases:
https://github.com/da-tubi/pants-airflow

And currently, I have to hack `sys.executable` to make it work:
``` python
from airflow.__main__ import main

def airflow_main():
    sys.executable = f"{AIRFLOW_HOME}/bin/python"
    main()
```

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
Correct. This is how airflow works, and we are not going to change it. Airflow webserver shoudl be using gunicorn, subprocess and forking and converting that into single packaged executable is not even remotely at the radar of any of the maintainers or generally speaking has never been discussed in the community.

If you wish to pursue your ideas related with packaging airflow. your best bet is raising a discussion at the devlist https://airflow.apache.org/community/  and explaining your motivation, and convincing the broader community that this is needed. Eventually that kind of expectation shoudl find it's way into https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals that you (or someone who picks it up) would descrbe, justify, work out all the details, go through voting process with successful vote and eventually implement most likely.

This is how architectural changes in Airflow are supposed to be performed. 

Converting it into discussion if more explanation is needed.
